src/detect-engine-hsbd.c

```diff
@@ -3631,6 +3847,308 @@ static int DetectEngineHttpServerBodyFileDataTest03(void)
     return result;
 }
 
+/*
+ * Tests buffer inspection like in ips mode,
+ * here all chunks size are smaller than the window_size.
+
+ * in:               buffer:
+ * [ab]              [ab]
+ * [cd]              [abcd]
+ * [ef]              [cdef]
+*/
+static int DetectEngineHttpServerBodyFileDataTest04(void)
+{
+
+    const char yaml[] = "\
+%YAML 1.1\n\
+---\n\
+libhtp:\n\
+\n\
+  default-config:\n\
+\n\
+    http-body-inline: yes\n\
+    response-body-minimal-inspect-size: 6\n\
+    response-body-inspect-window: 3\n\
+";
+
+    struct TestSteps steps[] = {
+        {   (const uint8_t *)"GET /index.html HTTP/1.0\r\n"
+            "Host: www.openinfosecfoundation.org\r\n"
+            "User-Agent: Mozilla/5.0 (X11; U; Linux i686; en-US; rv:1.9.1.7) Gecko/20091221 Firefox/3.5.7\r\n"
+            "\r\n",
+            0, STREAM_TOSERVER, 1, 0 },
+        {   (const uint8_t *)"HTTP/1.0 200 ok\r\n"
+            "Content-Type: text/html\r\n"
+            "Content-Length: 6\r\n"
+            "\r\n"
+            "ab",
+            0, STREAM_TOCLIENT, 1, 1 },
+        {   (const uint8_t *)"cd",
+            0, STREAM_TOCLIENT, 2, 1 },
+        {   (const uint8_t *)"ef",
+            0, STREAM_TOCLIENT, 3, 1 },
+        {   NULL, 0, 0, 0, 0},
+    };
+
+    const char *sigs[] = {
+        "alert http any any -> any any (file_data; content:\"ab\"; sid:1;)",
+        "alert http any any -> any any (file_data; content:\"abcd\"; sid:2;)",
+        "alert http any any -> any any (file_data; content:\"abcdef\"; sid:3;)",
+        NULL,
+    };
+
+    return RunTest(steps, sigs, yaml);
+}
+
+/*
+ * Tests buffer inspection like in ips mode,
+ * here the last chunk size is greater than the window size.
+
+ * in:               buffer:
+ * [ab]              [ab]
+ * [cd]              [abcd]
+ * [12345678]        [d12345678]
+*/
+static int DetectEngineHttpServerBodyFileDataTest05(void)
+{
+
+    const char yaml[] = "\
+%YAML 1.1\n\
+---\n\
+libhtp:\n\
+\n\
+  default-config:\n\
+\n\
+    http-body-inline: yes\n\
+    response-body-minimal-inspect-size: 6\n\
+    response-body-inspect-window: 3\n\
+";
+
+    struct TestSteps steps[] = {
+        {   (const uint8_t *)"GET /index.html HTTP/1.0\r\n"
+            "Host: www.openinfosecfoundation.org\r\n"
+            "User-Agent: Mozilla/5.0 (X11; U; Linux i686; en-US; rv:1.9.1.7) Gecko/20091221 Firefox/3.5.7\r\n"
+            "\r\n",
+            0, STREAM_TOSERVER, 1, 0 },
+        {   (const uint8_t *)"HTTP/1.0 200 ok\r\n"
+            "Content-Type: text/html\r\n"
+            "Content-Length: 13\r\n"
+            "\r\n"
+            "ab",
+            0, STREAM_TOCLIENT, 1, 1 },
+        {   (const uint8_t *)"cd",
+            0, STREAM_TOCLIENT, 2, 1 },
+        {   (const uint8_t *)"123456789",
+            0, STREAM_TOCLIENT, 3, 1 },
+        {   NULL, 0, 0, 0, 0},
+    };
+
+    const char *sigs[] = {
```

Comment:
I don't like that we're setting an expected result for just one sig per chunk, but we actually have 3 sigs in each test. I would prefer a method of passing expectations for each sig, perhaps in a separate array.